examples/simulation/plot_simulated_sample_dataset.py

```diff
@@ -0,0 +1,181 @@
+# in the events matrix.
+event_id = {'auditory/left': 1, 'auditory/right': 2, 'visual/left': 3,
+            'visual/right': 4, 'smiley': 5, 'button': 32}
+# Events from the experiment.
+events = mne.find_events(raw)
+# ##############################################################################
+
+# In order to simulate source time courses, labels of desired active regions
+# need to be specified for each of the 4 simulation conditions.
+# Make a dictionary that maps conditions to activation strengths within
+# aparc.a2009s labels. In the aparc.a2009s parcellation:
+#   -  'G_temp_sup-G_T_transv' is the label for primary auditory area 
+#   -  'S_calcarine' is the label for primary visual area
+# In each of the 4 conditions, only the primary area is activated. This means
+# that during the activations of auditory areas, there are no activations in
+# visual areas and vice versa.
+# Moreover, for each condition, contralateral region is more active (here, 2
+# times more) than the ipsilateral.
+
+activations = {
+    'auditory/left':
+        [('G_temp_sup-G_T_transv-lh', 100),          # label, activation (nAm)
+         ('G_temp_sup-G_T_transv-rh', 200)],
+    'auditory/right':
+        [('G_temp_sup-G_T_transv-lh', 200),
+         ('G_temp_sup-G_T_transv-rh', 100)],
+    'visual/left':
+        [('S_calcarine-lh', 100),
+         ('S_calcarine-rh', 200)],
+    'visual/right':
+        [('S_calcarine-lh', 200),
+         ('S_calcarine-rh', 100)],
+}
+
+annot='aparc.a2009s'
+
+# Load the 4 necessary label names.
+label_names = sorted(set(activation[0]
+                         for activation_list in activations.values()
+                         for activation in activation_list))
+region_names = list(activations.keys())
+
+#  Define the time course of the activity for each region to activate. We use a
+#  sine wave and it will be the same for all 4 regions.
+source_time_series = np.sin(np.linspace(0, 4 * np.pi, 100)) * 10e-9
+
+################################################################################
+# Create simulated source activity
+# --------------------------------
+
+# Here, SourceSimulator is used, which allows to specifiy where (label), what
+# (source_time_series), and when (events) event type will occur.
+
+# We will add data for 4 areas, each of which contains 2 labels. Since add_data
+# method accepts 1 label per call, it will be called 2 times per area.
+# All activations will contain the same waveform, but the amplitude will be 2
+# times higher in the contralateral label, as explained before.
+
+# When the activity occurs is defined using events. In this case, they are taken
+# from the original raw data. The first column is the sample of the event, the
+# second is not used. The third one is the event id, which is different for each
+# of the 4 areas.
+
+region_id = 0
+source_simulator = mne.simulation.SourceSimulator(src, tstep=tstep)
+for i, region_name in enumerate(region_names):
```

Comment:
```suggestion
for region_id, region_name in enumerate(region_names, 1):
```

then remove the use of `i`